### PR TITLE
Port of `yapapi.payload`

### PR DESCRIPTION
**Pull request by approxit**
What I've done:

* I've found errors in current `yapapi` design of classes (for e.g. unnecessary forcing `resolve_url` on manifest vm payload)
* In yapapi there was a concept of `Package`, which is removed here due to class hierarchy obfuscation
* Now object hierarchy is way clearer, due to fact of better naming decisions at removal of unnecessary stuff
* Old golem-core's `Payload.from_image_hash()` is now `VmPayload.from_repository()`
* Few functions related to repository resolving was revamped
* Changed all usage of old `Payload.from_image_hash` everywhere.

Also I would like to start introducing patterns to pull out hardcoded dependency like aiohttp.
As aiohttp client was used to resolve repository stuff, I've moved this to separate module, but my goal would be to give users ability to replace this with different set of tools of their choosing, if they desired this. 
One step closer to inversion of control and keeping our code more clean.

### CODE REVIEW

**Review by lucekdudek — APPROVED** [review]

**Comment by johny-b** [comment]
I haven't read his very carefully, but there's one change I really don't like :)

Imho code readability is much better with payload-generating function that is **not** async.

E.g. 
* I've done more than once `from somewhere import SomeSpecificPayload` - this is impossible now.
* Payload is nothing more than a config. It's nice to have `PAYLOAD = ...` at the top of the file. At least I like it :)

So. `Payload.from_image_hash` was obviously ugly, but:
1. Old `yapapi` `vm.repo` had no async internals (it was `async` because of legacy reasons, afair). Why can't we have same no-async interface now? Is it harmful in any way? Please note that if we create `Payload` **before** talking to `yagna` then sync reqest is not really a problem (or even before starting computations).
2. If we really want to avoid sync requests in `VmPayload.from_repository` then maybe this can be done in lazy way? E.g. execute async requests in `create_demand`?

File: golem_core/cli/utils.py [hunk 1]
```diff
@@ -79,7 +80,7 @@ def format_proposals(proposals: List[Proposal], first: bool) -> str:
 
 
 @dataclass
-class CliPayload(Payload):
+class CliPayload(BasePayload):
```
Comment (shadeofblue) [inline golem_core/cli/utils.py:83]: what's this? plus, why don't we have docstrings on classes?

File: golem_core/http_utils.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from typing import Optional
+
+import aiohttp
+
+# TODO: Make this module to be more like port / adapter thing for easily break out from aiohttp
```
Comment (shadeofblue) [inline golem_core/http_utils.py:5]: I don't understand this "todo" ...

**Review by shadeofblue — APPROVED** [review]
looks good, afaict :)

**Comment by approxit** [comment]
> I haven't read his very carefully, but there's one change I really don't like :)
> 
> Imho code readability is much better with payload-generating function that is **not** async.
> 
> E.g.
> 
> * I've done more than once `from somewhere import SomeSpecificPayload` - this is impossible now.
> * Payload is nothing more than a config. It's nice to have `PAYLOAD = ...` at the top of the file. At least I like it :)
> 
> So. `Payload.from_image_hash` was obviously ugly, but:
> 
> 1. Old `yapapi` `vm.repo` had no async internals (it was `async` because of legacy reasons, afair). Why can't we have same no-async interface now? Is it harmful in any way? Please note that if we create `Payload` **before** talking to `yagna` then sync reqest is not really a problem (or even before starting computations).
> 2. If we really want to avoid sync requests in `VmPayload.from_repository` then maybe this can be done in lazy way? E.g. execute async requests in `create_demand`?

PR was pivoted a little bit to address this issue, and with two options (keeping autodecoratingmodel stuff or removing it), we settle on keeping current approach to payloads. I've added `RepositoryVmPayload` that is fully declarative and it resolves `image_url` only at the stage of applying to demand. So looks like both worlds - fully declarative and with inline mappings payload->demand field (with props/constraints) - are happy.

**Review by shadeofblue — APPROVED** [review]

**Pull request merged**